src/Catel.Core/Catel.Core.Shared/Data/ModelBase.cs

```diff
@@ -626,6 +642,11 @@ protected void SetParent(IParent parent)
         /// </exception>
         public override bool Equals(object obj)
         {
+            if (DisableUsageOfEqualityComparerForEquals)
+            {
+                return RuntimeHelpers.Equals(this, obj);
```

Comment:
Use ObjectHelper.AreEqualReferences instead if possible. Then we can always improve the performance of that method instead of RuntimeHelpers.Equals performance better